### ISSUE TITLE
NetworkInterface using ApplicationGatewayBackendAddressPoolsIds is resulting in unexpected null.

### ISSUE THREAD

**Issue by JFenner-Evans** [pulumi/pulumi#2740]
I have created a new NetworkInterface which has, within it's ipConfiguration, some applicationGatewayBackendAddressPoolsIds referencing to the new ApplicationGateway I have also created in the stack.

ApplicationGateway:
`export const appGateway = new azure.network.ApplicationGateway("agw", {
        name: "agw",
        backendAddressPools: [{name: "appGatewayBackendPool"}],
...`

NetworkInterface:
`const nic = new azure.network.NetworkInterface("nic", {
                ipConfigurations: [{
                    name: "main",
                    privateIpAddressAllocation: "Dynamic",
                    applicationGatewayBackendAddressPoolsIds: [pulumi.output(network.appGateway.backendAddressPools[0].id)],
...`

I have made NetworkInterface depend on appGateway to ensure its dependency as well.
However, when I "pulumi up" it fails on the NetworkInterface and returns with: 
`error: unexpected null property application_gateway_backend_address_pools_ids[0]`
I assume this is because of the id of the backendPool being a string and not a pulumi Output?

So in summary, my question is: How can I create an Application Gateway with a Backend Pool and include it in my Network Interface?

**Comment by leezen** [comment]
I'm not able to reproduce this issue. Please re-open if you're still having trouble. Thanks!
